src/backend/vulkan/src/lib.rs

```diff
@@ -1476,6 +1476,116 @@ impl queue::CommandQueue<Backend> for CommandQueue {
         assert_eq!(Ok(()), result);
     }
 
+    unsafe fn bind_sparse<'a, M, Bf, I, S, Iw, Is, Ibi, Ib, Iii, Io, Ii>(
+        &mut self,
+        info: queue::BindSparseInfo<Iw, Is, Ib, Io, Ii>,
+        device: &Device,
+        fence: Option<&native::Fence>,
+    ) where
+        Bf: 'a + BorrowMut<native::Buffer>,
+        M: 'a + Borrow<native::Memory>,
+        Ibi: IntoIterator<Item = queue::SparseMemoryBind<&'a M>>,
+        Ib: IntoIterator<Item = (&'a mut Bf, Ibi)>,
+        I: 'a + BorrowMut<native::Image>,
+        Iii: IntoIterator<Item = queue::SparseImageMemoryBind<'a, &'a M>>,
+        Io: IntoIterator<Item = (&'a mut I, Ibi)>,
+        Ii: IntoIterator<Item = (&'a mut I, Iii)>,
+        S: 'a + Borrow<native::Semaphore>,
+        Iw: IntoIterator<Item = &'a S>,
+        Is: IntoIterator<Item = &'a S>
+    {
+        //TODO: avoid heap allocations
+        let mut waits = Vec::new();
+
+        for semaphore in info.wait_semaphores {
+            waits.push(semaphore.borrow().0);
+        }
+        let signals = info
+            .signal_semaphores
+            .into_iter()
+            .map(|semaphore| semaphore.borrow().0)
+            .collect::<Vec<_>>();
+
+        let mut buffer_memory_binds = Vec::new();
+        let buffer_binds = info.buffer_memory_binds.into_iter().map(|(buffer, bind_iter)| {
+            let binds_before = buffer_memory_binds.len();
+            buffer_memory_binds.extend(bind_iter.into_iter().map(|bind| {
```

Comment:
With there being nested iterators, it will be pretty hard to get that `total_buffer_binds`. A maybe easier implementation would be to use a vector for each binding iterator and drop everything in the end, but it would make the heap allocation problem worse. I'll try to think of something